### ISSUE TITLE
Documentation: Import JuliaPraxis guidelines.

### ISSUE THREAD

**Issue by Ismael-VC** [JuliaLang/julia#25204]
JuliaPraxis is an organization that emerged from the need to talk about "Julia: Best Practices", there has been lots of discussions and some advances but recently it has been brought to my attention, that the current JuliaPraxis effort could be potentially ignored, because those guidelines are not part of this repository.

* https://github.com/JuliaPraxis
* https://gitter.im/JuliaPraxis (chat)

So I think we need to start importing those guidelines over here while discussing them in order to reach consensus on which could be considered "official" guidelines, so we can enforce them later with the aim to make the language more consistent, so others can follow the example laid out in this repository for the Julia code style and conventions.

To do:
- [ ] Bike shed cryptic top level identifier names (starting by priority with exported ones, but also including non exported ones.)
- [ ] Bike shed keyword argument names (:warning: **breaking**, would need to be done before 1.0, but we already have code freeze.)
- [ ] [Bike shed better positional argument names post 1.0](https://github.com/JuliaLang/julia/issues/25198)

Related (mostly all the "rename x" open issues), but style also includes any kind of "refactoring tips" found in other issues, or just "best practices" in general (including for example things like documenting and coverage practices too.) Here a list of the most recent ones:

* https://github.com/JuliaLang/julia/issues/25104
* https://github.com/JuliaLang/julia/issues/25082
* https://github.com/JuliaLang/julia/issues/24967
* https://github.com/JuliaLang/julia/issues/24910
* https://github.com/JuliaLang/julia/issues/25118
* https://github.com/JuliaLang/julia/issues/24865
* https://github.com/JuliaLang/julia/issues/24910
* https://github.com/JuliaLang/julia/issues/24815
* https://github.com/JuliaLang/julia/issues/24645
* https://github.com/JuliaLang/julia/issues/23710
* https://github.com/JuliaLang/julia/issues/23502
* https://github.com/JuliaLang/julia/issues/23482
* https://github.com/JuliaLang/julia/issues/23022
* https://github.com/JuliaLang/julia/pull/25105
* https://github.com/JuliaLang/julia/pull/25100
* https://github.com/JuliaLang/julia/pull/12706
* https://github.com/JuliaLang/julia/pull/12350

In the cases where JuliaPraxis suggestions are deemed inappropriate, such suggestions should be replaced with appropriate ones we can agree from the stylistic and naming patterns currently in use here.

So that all the issues/PRs that reflect agreement about any type of stylistic convention, should be properly documented along with the other guidelines, for the sake of consistency.

Priority would be naming style and conventions, since those are user-facing (part of APIs), and would extend to syntactic, workflow, tips, etc. type of guidelines.
